packages/schema/parse/src/extract/utils.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  createEnumDefinition,
+  createObjectDefinition,
+  createScalarDefinition,
+  isScalarType,
+  PropertyDefinition,
+} from "../typeInfo";
+import { isEnumType } from "../typeInfo/enum";
+import { isObjectType } from "../typeInfo/object";
+import { TypeDefinitions } from "./type-definitions";
+
+export function setPropertyType(
+  property: PropertyDefinition,
+  name: string,
+  type: { type: string; required: boolean | undefined },
+  typeDefinitions: TypeDefinitions
+): void {
+  if (isScalarType(type.type)) {
+    property.scalar = createScalarDefinition({
+      name: name,
+      type: type.type,
+      required: type.required,
+    });
+  } else if (isEnumType(type.type, typeDefinitions.enumTypes)) {
```

Comment:
Ideally we should be able to do this by looking at the GraphQL type's "kind" property.